locations/spiders/alnatura_de.py

```diff
@@ -0,0 +1,90 @@
+import scrapy
+import re
+import json
+
+from locations.items import GeojsonPointItem
+from locations.hours import OpeningHours
+
+
+DAY_MAPPING = {
+    1: 'Mo', 2: 'Tu', 3: 'We', 4: 'Th', 5: 'Fr', 6: 'Sa', 7: 'Su',
+    'Mo': 1, 'Tu': 2, 'We': 3, 'Th': 4, 'Fr': 5, 'Sa': 6, 'Su': 7
+}
+
+
+class AlnaturaSpider(scrapy.Spider):
+    name = "alnatura_de"
+    allowed_domains = ["www.alnatura.de"]
+    start_urls = (
+        'https://www.alnatura.de/api/sitecore/stores/FindStoresforMap?'
+        'ElementsPerPage=10000&lat=50.99820058296841'
+        '&lng=7.811966062500009&radius=1483'
+        '&Tradepartner=Alnatura%20Super%20Natur%20Markt',
+    )
+
+    def parse_hours(self, store_hours):
+        opening_hours = OpeningHours()
+        match = re.match(r'(.+?)-(.+?) +(\d.*?)-(.+?) Uhr', store_hours)
+        if match:
+            from_day = match.group(1).strip()
+            to_day = match.group(2).strip()
+            from_time = match.group(3).strip()
+            to_time = match.group(4).strip()
+
+            fhours = int(float(from_time))
+            fminutes = (float(from_time) * 60) % 60
+            fmt_from_time = "%d:%02d" % (fhours, fminutes)
+            thours = int(float(to_time))
+            tminutes = (float(to_time) * 60) % 60
+            fmt_to_time = "%d:%02d" % (thours, tminutes)
+
+        for day in range(DAY_MAPPING[from_day], DAY_MAPPING[to_day] + 1):
+            opening_hours.add_range(
+                day=DAY_MAPPING[day],
+                open_time=fmt_from_time,
+                close_time=fmt_to_time,
+                time_format='%H:%M'
+            )
+
+        return opening_hours.as_opening_hours()
+
+    def parse_stores(self, response):
+        store = json.loads(response.text)
+        store = store['Payload']
+
+        properties = {
+            'lat': response.meta.get('lat'),
+            'lon': response.meta.get('lng'),
+            'name': store['StoreName'],
+            'street': store['Street'],
+            'city': store['City'],
+            'postcode': store['PostalCode'],
+            'phone': store['Tel'],
+            'country': store['Country'],
+            'ref': response.meta.get('id'),
+            'extras': {
+                'time': store['OpeningTime']
+            }
+        }
+
+        if store['OpeningTime']:
```

Comment:
You might want to use `store.get('OpeningTime')` so that you get a `None` instead of a `KeyError` here if a store doesn't have this key in it.

```suggestion
        if store.get('OpeningTime'):
```

Reply:
Done.